src/cocoa_window.m

```diff
@@ -854,6 +855,17 @@ static GLFWbool createNativeWindow(_GLFWwindow* window,
         [window->ns.object setTabbingMode:NSWindowTabbingModeDisallowed];
 #endif
 
+    // EGL implementation on macOS use CALayer* EGLNativeWindowType so we need
+    // to get the layer for EGL window surface creation.
+    if (ctxconfig->source == GLFW_EGL_CONTEXT_API)
+    {
+        [window->ns.view setWantsLayer:YES];
+        window->ns.layer = [window->ns.view layer];
+
+        if (window->ns.retina)
+            [window->ns.layer setContentsScale:[window->ns.object backingScaleFactor]];
```

Comment:
Did the layer not pick up the contents scale from the view without this?

Reply:
It does, removed.